src/hpack/decoder.rs

```diff
@@ -495,14 +497,13 @@ impl Table {
     }
 
     fn reserve(&mut self, size: usize) {
-        debug_assert!(size <= self.max_size);
-
         while self.size + size > self.max_size {
-            let last = self.entries
-                .pop_back()
-                .expect("size of table != 0, but no headers left!");
-
-            self.size -= last.len();
+            match self.entries.pop_back() {
+                Some(last) => {
+                    self.size -= last.len();
+                }
+                None => return,
```

Comment:
this return is surprising because there is nothing else in the function? not fundamentally opposed to this, just noting

Reply:
It's for breaking out of the loop.

Reply:
Why not `break`?

Reply:
IMO it makes it more obvious that, at that point, the fn is done vs. having to find the end of the loop and see what happens there.

Just a general opinion. The fn is small.

Reply:
Yeah, that makes sense, just curious!